frameworks/elastic/tests/config.py

```diff
@@ -167,46 +180,79 @@ def get_elasticsearch_master(service_name=SERVICE_NAME):
         "vertices": [{"field": "name"}],
         "connections": {"vertices": [{"field": "role"}]},
     }
 
-    # The graph endpoint doesn't exist without X-Pack installed.
-    # In that case Elasticsearch returns a plain text error:
-    # "No handler found for uri [/INDEX_NAME/_xpack/graph/explore] and method [POST]"
-    response = _curl_query(
-        service_name,
-        "POST",
-        "{}/_xpack/_graph/_explore".format(DEFAULT_INDEX_NAME),
-        json_data=query,
-        return_json=is_enabled,
+    response = explore_graph(
+        service_name, index_name, query, http_user=http_user, http_password=http_password
     )
-    if is_enabled:
-        return is_graph_endpoint_active(response)
-    else:
-        return "No handler found" in response
 
+    delete_index(
+        index_name, service_name=service_name, http_user=http_user, http_password=http_password
+    )
 
-# The graph endpoint response looks something like:
-# {
-#   "took": 200,
-#   "timed_out": false,
-#   "failures": [],
-#   "vertices": [],
-#   "connections": []
-# }
-def is_graph_endpoint_active(response):
-    return isinstance(response["vertices"], list) and isinstance(response["connections"], list)
+    return is_expected_to_be_enabled == is_graph_explore_endpoint_active(response)
 
 
-def set_xpack(is_enabled, service_name=SERVICE_NAME):
-    # Toggling X-Pack requires full cluster restart, not a rolling restart
-    options = {"TASKCFG_ALL_XPACK_ENABLED": str(is_enabled).lower(), "UPDATE_STRATEGY": "parallel"}
-    update_app(service_name, options, DEFAULT_TASK_COUNT)
+def verify_commercial_api_status(
+    is_expected_to_be_enabled, service_name=SERVICE_NAME, http_user=None, http_password=None
+):
+    return verify_graph_explore_endpoint(
+        is_expected_to_be_enabled, service_name, http_user=http_user, http_password=http_password
+    )
+
+
+# On Elastic 6.x, the "Graph Explore API" is available when the Elasticsearch cluster is configured
+# with a "trial" X-Pack license. When configured with a "basic" license (the default) the API will
+# respond with an HTTP 403.
+#
+# A "Graph Explore API" response will look something like:
+#   1. With a "trial" license:
+#   {
+#     "took": 183,
+#     "timed_out": false,
+#     "failures": [],
+#     "vertices": [...],
+#     "connections": [...]
+#   }
+#
+#   2. With a "basic" license:
+#   {
+#     "error": {
+#       "root_cause": [
+#         {
+#           "type": "security_exception",
+#           "reason": "current license is non-compliant for [graph]",
+#           "license.expired.feature": "graph"
+#         }
+#       ],
+#       "type": "security_exception",
+#       "reason": "current license is non-compliant for [graph]",
+#       "license.expired.feature": "graph"
+#     },
+#     "status": 403
+#   }
+def is_graph_explore_endpoint_active(response):
+    return isinstance(response.get("vertices"), list) and isinstance(
+        response.get("connections"), list
```

Comment:
It would, but checking that both `connections` and `vertices` are present in the response is a better way to show that the intent here is to verify that this is a response from the graph endpoint.